ssh/tailssh/user.go

```diff
@@ -74,11 +74,18 @@ func userLookup(uid string) (*userMeta, error) {
 }
 
 func validUsername(uid string) bool {
-	if len(uid) > 32 || len(uid) == 0 {
+	maxUid := 32
+	if runtime.GOOS == "linux" {
+		maxUid = 256
+	}
```

Comment:
this part seems like a fine change but unrelated? send as its own PR?

Reply:
Will revert and open separate PR.

Reply:
https://github.com/tailscale/tailscale/pull/8278